Assets/Scripts/Game Controllers/LevelEndTriggerController.cs

```diff
@@ -3,11 +3,34 @@
 
 public class LevelEndTriggerController : MonoBehaviour
 {
+    [SerializeField]
+    private int indexForNextScene;
+
+    private const int lastLevel = 2;
+
+    private void Start()
+    {
+        indexForNextScene = SceneManager.GetActiveScene().buildIndex + 1;
+    }
+
     private void OnTriggerEnter2D(Collider2D collision)
-    {   
-        if(collision.CompareTag("Player"))
+    {
+        if (collision.CompareTag("Player"))
         {
-            SceneManager.LoadScene("LevelSelect");
+            if ((SceneManager.GetActiveScene().buildIndex - 1) == lastLevel)
```

Comment:
This condition should be `SceneManager.GetActiveScene().buildIndex - 1 == lastLevel` because of the level scene build index change.